### ISSUE TITLE
`name-replacements`: allow `ProcessEnv` by default

### ISSUE THREAD

**Issue by chatoo2412** [sindresorhus/eslint-plugin-unicorn#3605]
### Description

`@types/node` declares the interface as `NodeJS.ProcessEnv`, so giving `process.env` a type requires declaration merging against an interface with exactly that name. Following the rule's suggestion doesn't rename anything — it declares an unrelated `ProcessEnvironment` interface, the augmentation stops applying, and every `process.env.*` access falls back to `string | undefined`. There is no way to comply with the rule and keep the typing.

This is the same category as `ImportMetaEnv`, which was added to `defaultAllowList` after #3371: a name whose spelling is fixed by an external declaration. `ProcessEnv` has wider reach than `ImportMetaEnv` — it applies to any Node project that types its environment variables, not just Vite projects.

**Suggested change**: add `ProcessEnv: true` to `defaultAllowList` in `rules/shared/name-replacements.js`, next to `ImportMetaEnv` and `EmberENV`.

References: [`process.env`](https://nodejs.org/api/process.html#processenv), [the `NodeJS.ProcessEnv` declaration in `@types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/process.d.ts).

### Examples

```ts
// ✅ should not be reported: the name is fixed by `@types/node`
declare namespace NodeJS {
	interface ProcessEnv {
		FILE_PATH: string;
	}
}
```

```ts
// ❌ what the rule currently asks for: this declares an unrelated interface,
// so `process.env.FILE_PATH` goes back to `string | undefined`
declare namespace NodeJS {
	interface ProcessEnvironment {
		FILE_PATH: string;
	}
}
```

Current output:

```
error  The variable `ProcessEnv` should be named `ProcessEnvironment`. A more descriptive name will do too  unicorn/name-replacements
```

### Additional Info

eslint-plugin-unicorn 73.0.0, eslint 10.8.0, typescript 6.0.3, `unicorn/recommended` with no other options. Currently worked around with `allowList: {ProcessEnv: true}`.
